controllers/openstackdataplane_controller.go

```diff
@@ -412,3 +419,13 @@ func buildBMHHostMap(instance *dataplanev1beta1.OpenStackDataPlane,
 		}
 	}
 }
+
+// checkDeployProtectionAnnotation checks to see if the node has the deploy protection annotation set. If this is set,
+// we will block the execution of Ansible tasks until it is cleared.
+func (r *OpenStackDataPlaneReconciler) checkDeployProtectionAnnotation(instance *dataplanev1beta1.OpenStackDataPlane, annotation map[string]string) bool {
+	for k := range instance.ObjectMeta.GetAnnotations() {
+		_, keyExists := annotation[k]
+		return keyExists
```

Comment:
This now does not check the value of the annotation. So we have to carefully document that fact. E.g `"openstack.org/deploy-protection": "false"`, does not mean the protection is disabled.

Reply:
I think this also needs to check if keyExists and its value is true'ish or an empty string.

Reply:
This was intentional. I thought it would ensure there is definitely explicit intent if the user has to completely remove the annotation rather than just changing it to false. But if we think it should accept both true and false, we can change it also compare values.